numpy/core/include/numpy/npy_cpu.h

```diff
@@ -115,4 +116,83 @@
 #define NPY_CPU_HAVE_UNALIGNED_ACCESS 0
 #endif
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum
+{
+    NPY_OPT_NONE = 0,
+    /* X86 */
+    NPY_OPT_MMX               = 1,
```

Comment:
Is this a standard approach you took from somewhere else? The numbers do feel a bit magical......

Reply:
I preferred to follow OpenCV style for obvious reasons, feature's values should be unique and unchangeable because we gonna use it later as entries to check/disable/enable dispatch CPU features on runtime via environment variables or via python API, also the gap between values could be useful for verify entries, i.e `(NPY_OPT_NONE > x && x < PY_OPT_VSX) == X86`

Reply:
Okay, so this is what OpenCV does and it's been proven. Perhaps it just needs a bit more commenting above with context and how these are used. The bit shifting code for these values isn't exactly the easiest to understand when reading for the first time.

Reply:
I think you're right but even though the reader have to read definition of `npy_cpu_have` and `npy_cpu_init` to fully understand how theses are used, BTW usually there's no need to use it directly, it always gonna used behind the seen inside macros or functions, i.e in this patch all you need is just call macro `NPY_CPU_HAVE(FEATURE NAME)`, NPY_CPU_HAVE(AVX2) instead of `npy_cpu_have(NPY_OPT_AVX2)`.

however I will try to improve documentation on a separate patch  but after I get done from implement universal intrinsics.

by the way I'm made a mistake in here I didn't mean to leave a gap between `NPY_OPT_SSE41` and `NPY_OPT_POPCNT`, I'm going to fix it

Reply:
For the record, the gap between feature IDs left only for ABI compatibility and there's no intention to use it from outside. Only strings.